### PR TITLE
Improved check of signatures subset in stateful validator

### PR DESCRIPTION
**Pull request by darsh7807**
### Description of the Change
I made two loops to check the whether the public key of signatures exists in the public_keys.

### Benefits
This is making code readable 

### Possible Drawbacks 
It might slow down the runtime of the function

### CODE REVIEW

**Comment by neewy** [comment]
@darsh7807 
> It might slow down the runtime of the function

Can you please tell more?

**Comment by darsh7807** [comment]
@neewy Previous implementation was using set to store keys which is O(nlogn) operation. all_of()  function uses O(n) 
So,thats O(nlogn) complexity.
My implementation is O(n*m) so it can slow down for larger data.

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 1]
```diff
@@ -98,16 +98,16 @@ namespace iroha {
         const std::vector<shared_model::crypto::PublicKey> &public_keys) {
       // TODO 09/10/17 Lebedev: simplify the subset verification IR-510
```
Comment (l4l) [inline irohad/validation/impl/stateful_validator_impl.cpp]: remove the comment as well

Comment (darsh7807) [reply irohad/validation/impl/stateful_validator_impl.cpp]: Okay I have removed the comments

**Comment by lebdron** [comment]
@darsh7807 According to https://github.com/hyperledger/iroha/issues/628 , the idea was to improve the efficiency in terms of algorithmic runtime complexity. Since current approach has greater runtime complexity, it does not satisfy the requirements described in the issue.

**Comment by darsh7807** [comment]
@lebdron  We discussed this on Iroha channel on the telegram and what I was informed was the code should be more easy to read and the complexity was not a big issue. So I increased the complexity. 
You can see my attempt to improve complexity in the issue comments but on a channel, they gave feedback that it was not more readable and they wanted a good readable solution.

**Comment by lebdron** [comment]
@darsh7807 Unfortunately there was a miscommunication in developer chat, and no comments were provided on the proposed solution. After revision of issue solutions requirements, we came to conclusion that `O(n^2)` might be unacceptable if we increase the maximum number of signatures allowed for transactions and accounts.

**Comment by darsh7807** [comment]
@lebdron , okay it's fine then.  I will propose another solution using unordered  map which I was discussing on channel. I am out for 9 days. I will submit the solution once I am back.

**Comment by lebdron** [comment]
Please reopen the pull request when it will be completed.

**Pull request closed without merging**